contracts/BondStorage.sol

```diff
@@ -18,13 +19,15 @@ contract BondStorage is AccessControl {
     uint8 public otherUsdDec;
 
     constructor(address _gatewayAddress, address _chainlinkEurOther, uint8 _otherUsdDec) {
-        _setupRole(WHITELIST_BOND_STORAGE, msg.sender);
+        _grantRole(WHITELIST_ADMIN, msg.sender);
+        _setRoleAdmin(WHITELIST_BOND_STORAGE, WHITELIST_ADMIN);
```

Comment:
this means that we give the `BondStorage` contract owner the admin role over the `WHITELIST_BOND_STORAGE` role.

what are your thoughts about centralizing even more power to the operator versus giving the `BondStorage` more autonomous power (as in this case)?

Reply:
as i understand, the contract itself was the admin of its own whitelist role, there would be no signer to call the transaction and grant and revoke those roles. a contract can't grant a role itself, right?